eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerDeleteById.kt

```diff
@@ -18,19 +20,32 @@ class AssignmentNatsControllerDeleteById(
     override val subject: String = ASSIGNMENT_DELETE_BY_ID
     override val parser: Parser<DeleteByIdAssignmentRequest> = DeleteByIdAssignmentRequest.parser()
 
-    override fun handler(request: DeleteByIdAssignmentRequest): DeleteByIdAssignmentResponse =
-        runCatching {
-            val id = request.request.assignmentId.id
-            service.delete(id).block()
-            getSuccessResponse()
-        }.getOrElse { ex ->
-            getFailureResponse(ex.javaClass.simpleName, ex.toString())
+    override fun handler(request: DeleteByIdAssignmentRequest): Mono<DeleteByIdAssignmentResponse> {
+        val assignmentRequest = request.request
+
+        return when (assignmentRequest.requestCase) {
+            AssignmentRequest.RequestCase.ASSIGNMENT_ID ->
+                Mono.defer {
+                    val id = assignmentRequest.assignmentId.id
+                    service.delete(id).thenReturn(getSuccessResponse())
+                }.onErrorResume {
+                    Mono.just(getFailureResponse(it.javaClass.simpleName, it.toString()))
+                }
+
+            AssignmentRequest.RequestCase.ASSIGNMENT ->
+                Mono.just(getFailureResponse("UnsupportedRequestType", "AssignmentProto is not supported for deletion"))
+
+            AssignmentRequest.RequestCase.REQUEST_NOT_SET ->
+                Mono.just(getFailureResponse("InvalidRequest", "No request type set within AssignmentRequest"))
+
+            else ->
+                Mono.just(getFailureResponse("UnknownRequestType", "Unknown request type within AssignmentRequest"))
```

Comment:
let's handle all possible cases (without else branch). So, when somebody adds a new case to oneof, he is obliged by the compilator to add handling of his case. fix in all cases

Reply:
`+`